src/System.Runtime.Extensions/src/System/Environment.Win32.cs

```diff
@@ -219,7 +219,7 @@ private static string GetKnownFolderPath(string folderGuid, SpecialFolderOption
             {
                 if (hr == Interop.Shell32.COR_E_PLATFORMNOTSUPPORTED)
                 {
-                    throw new PlatformNotSupportedException();
+                    throw new PlatformNotSupportedException(SR.NotSupported);
```

Comment:
I couldn't figure out how SHGetKnownFolderPath would lead to COR_E_PLATFORMNOTSUPPORTED. Do you know?